### ISSUE TITLE
Speech in EPUB's SVGs?

### ISSUE THREAD

**Issue by rbeezer** [PreTeXtBook/pretext#1778]
When building EPUB with SVG images, we "enrich" the SVG with speech strings placed in a standard SVG `desc` element.

On the one hand, this makes Speech Rule Engine (SRE) a dependency for EPUB.  On the other hand, MathJax is required no matter what, so the use of `npm` is necessary, and we have a small script that installs both.  Perhaps this is only an issue if the CLI was to ship the necessary `node_modules` directory, etc, to support MJ and SRE?

One solution is a new format in the core routines.  For the existing format `epub-svg` remove the speech enhancement, and then add a new format, say `epub-enhanced-svg` that includes the SRE work.

How does the CLI feel about EPUB support these days?  A new format is do-able, but only if we (collectively) feel it is necessary (generally, @mitchkeller), or for the CLI (@oscarlevin @StevenClontz).

Line in question:
https://github.com/PreTeXtBook/pretext/blob/8fe4148da635a7b4e64103815d7c07cfe16cb352/pretext/pretext.py#L2063

Commit to guide any changes: 28d80dea9447ae104abb2315caf63a7ddec4b7e2

**Comment by StevenClontz** [comment]
@oscarlevin has thought much more carefully about epub than I have so I'll defer to his perspective here.

**Comment by oscarlevin** [comment]
TLDR; I don't think it is worth creating a new format just to make the CLI easier.  We will want a solution that includes SRE anyway because we will want to support braille.

The way we have everything set up now means all core pretext stuff is installed in a static folder by pip that doesn't allow for the user to npm install to.  So either we restructure all the support files to be installed into the user directory (which might be something we want to do eventually) or we ship the node_modules folders.  Assuming we are allowed to do that (I don't know what the licensing rules are here), then it doesn't matter much to me whether we ship the sre node_modules in addition (it is about 4MB, which is way larger than the rest of them, but for now, I don't think unreasonable, right?).  

There is still work to be done before I'm confident we can include epub, including figuring out why it doesn't work on windows (there is a permission error when using the "inplace" fix for math, apparently).  I'm tracking those issues here: https://github.com/PreTeXtBook/pretext-cli/issues/286.

**Comment by StevenClontz** [comment]
See also https://github.com/PreTeXtBook/pretext-cli/issues/180 Since our XSLT only works if it lives on the disk, and that's not actually guaranteed by Python packaging spec (e.g. could be distributed in a compressed tarball) we should be "installing" files, I think to `~/.ptx/1.2.3/`. That's also where we could install `node_modules`. 

I think I prefer shipping a `package.json` to `npm install` rather than redistributing code directly.

**Comment by oscarlevin** [comment]
I'm definitely leaning more that way now, and it makes the case for not introducing a new format even stronger.

**Comment by mitchkeller** [comment]
I think as long as the EPUB that is being produced is valid, it's not a problem to include the speech strings in the SVGs. With node required either way, it seems like it's sensible to stick with making the more accessible EPUB by including speech rather than creating yet another EPUB variant.
